### PR TITLE
When given a circular object to send to spoor, throw a custom error message that states circular objects are not supported 

### PR DESCRIPTION
**Pull request by JakeChampion**
Resolves https://github.com/Financial-Times/o-tracking/issues/115

After speaking with the Data team and Tom Parker, we have come to the conclusion that circular objects are not going to be supported in Spoor and that o-tracking should throw an error if given a circular object

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to 7a426ec3d00649e35bc1ce5d53c362e551ae0ac0
js: 5.62kb increase (1.91kb/gzip)

File: bower.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
   ],
   "description": "Origami module for FT tracking.",
   "dependencies": {
-    "ftdomdelegate": "^4.0.0"
+    "ftdomdelegate": "^4.0.0",
+    "is-circular": "tjmehta/is-circular#^1.0.2"
```
Comment (notlee) [inline bower.json:15]: `tjmehta/is-circular` is quite a small package without many contributors. For security maybe we should implement directly in `o-tracking` or lock down to a specific version?

File: src/javascript/utils.js [hunk 2]
```diff
@@ -253,6 +253,51 @@ function assignIfUndefined (subject, target) {
 	}
 }
 
+/**
+ * Used to find out all the paths which contain a circular reference.
+ * @param {*} rootObject The object we want to search within for circular references
+ * @returns {string[]} Returns an array of strings, the strings are the full paths to the circular references within the rootObject
+ */
+function findCircularPathsIn(rootObject) {
+	const traversedValues = new WeakSet();
```
Comment (notlee) [inline src/javascript/utils.js:262]: Users would need a new polyfill for `WeakSet`so this is a breaking change

Comment (JakeChampion) [reply src/javascript/utils.js:262]: I changed the base branch to merge into v3 instead

File: bower.json (hunk 1 above)
Comment (JakeChampion) [reply bower.json:15]: Sure thing, I coded one up myself but decided to use their version instead for some reason --
https://gist.github.com/JakeChampion/e1c593b67c59d0f75c7ca3c1cb86a21f

File: src/javascript/utils.js (hunk 2 above)
Comment (notlee) [reply src/javascript/utils.js:262]: cool :) don't forget to add it to `origami.json`!

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to cb743ab1919be546c1b6bc8529ffd46f36b643c4
js: 5.76kb increase (1.88kb/gzip)

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to 080d5713e35ef0c4e29d462fff2a3092094f8a0e
js: 5.76kb increase (1.88kb/gzip)

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to 93fba8280f151ba29548d35115de75a1050561c7
js: 5.48kb increase (1.78kb/gzip)

**Comment by JakeChampion** [comment]
@notlee please take a look again, I've now updated the migration guide and origami.json.browserFeatures.required array and swapped out is-circular for our own implentation

**Pull request merged**